pkg/sensors/exec/exec_test.go

```diff
@@ -263,9 +263,13 @@ func TestEventExecve(t *testing.T) {
 
 	myCaps := ec.NewCapabilitiesChecker().FromCapabilities(caps.GetCurrentCapabilities())
 	myNs := ec.NewNamespacesChecker().FromNamespaces(namespace.GetCurrentNamespace())
-
+	rootAccount := ec.NewUserAccountChecker()
+	rootAccount.User = ec.NewUserRecordChecker().FromUserRecord(&tetragon.UserRecord{
+		Name: "root",
+	})
 	procChecker := ec.NewProcessChecker().
 		WithBinary(sm.Full(testNop)).
+		WithAccount(rootAccount).
```

Comment:
Seems this test is failing?

Let's separate things so we don't affect other workflow, create a new test  TextEventExecveWithUsername()

Can you please try this and see if the test passes on your local setup? then we can go with it:
*  Inside test do https://github.com/cilium/tetragon/blob/8100230e93aa5d86b9a8ca863e1d1e165d06ce53/pkg/sensors/tracing/policyfilter_test.go#L111...L113  to initialize detection
* Do the same check as in the code to confirm if it is running in host and test systemd deployment for test, if it fails skip test.
* Otherwise continue with observertesthelper.GetDefaultObserver() and  assert that username root is set

If it works please do also check the exit event with username, so we ensure it is cached

Reply:
Could you explain please, why do we need this: https://github.com/cilium/tetragon/blob/8100230e93aa5d86b9a8ca863e1d1e165d06ce53/pkg/sensors/tracing/policyfilter_test.go#L111...L113?
I made a separate test with namespace check and ExitEvent check. It works fine locally without this initialization.

Reply:
Great then! the initialization allows to discover cgroup , deployment etc! so if you want to do the GetDeploymentMode() we need to initialize it first.

Doing the namespace check if in host mnt/user is fine for tests only, as right now they run directly, but in main tetragon we need to detect the systemd deployment as we may not control the environment.